README.md

```diff
@@ -1,8 +1,137 @@
-# Tauri Plugin oauth
+# Tauri Plugin OAuth
 
-WIP
+![WIP Badge](https://img.shields.io/badge/status-WIP-yellow)
 
-Minimalistic rust library and Tauri plugin(soon) to spawn a temporary localhost server which you redirect to from browser based oauth flows ("Login with X").
-Needed because many sites such as Google and GitHub don't allow custom URI schemes ("deep link") as redirect URLs.
+A minimalistic Rust library and Tauri plugin for handling browser-based OAuth flows in desktop
+applications. This plugin spawns a temporary localhost server to capture OAuth redirects, solving
+the challenge of using OAuth with desktop apps.
 
-See https://github.com/FabianLars/tauri-plugin-deep-link for an alternative based on deep linking. This one will automatically start your app if there is no open instance.
+## Why This Plugin?
+
+Many OAuth providers (like Google and GitHub) don't allow custom URI schemes ("deep links") as
+redirect URLs for security reasons. This plugin provides a solution by:
```

Comment:
```suggestion
redirect URLs. This plugin provides a solution by:
```
tbh feels more like most of them don't support it just because. I'd prefer not to say it's for security reasons so that i'm not asked to explain what those are 😅